linera-base/src/identifiers.rs

```diff
@@ -574,6 +584,89 @@ impl std::str::FromStr for Owner {
     }
 }
 
+impl Serialize for AccountOwner {
+    fn serialize<S: serde::ser::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&self.to_string())
+        } else {
+            match *self {
+                AccountOwner::Application(app_id) => {
+                    serializer.serialize_newtype_variant("AccountOwner", 0, "Application", &app_id)
+                }
+                AccountOwner::User(owner) => {
+                    serializer.serialize_newtype_variant("AccountOwner", 1, "User", &owner)
+                }
+            }
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for AccountOwner {
+    fn deserialize<D: serde::de::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        if deserializer.is_human_readable() {
+            let s = String::deserialize(deserializer)?;
+            let value = Self::from_str(&s).map_err(serde::de::Error::custom)?;
+            Ok(value)
+        } else {
+            #[derive(Deserialize)]
+            #[serde(rename = "AccountOwner")]
+            enum Foo {
+                User(Owner),
+                Application(ApplicationId),
+            }
+
+            let value = Foo::deserialize(deserializer)?;
+            match value {
+                Foo::Application(app_id) => Ok(AccountOwner::Application(app_id)),
+                Foo::User(owner) => Ok(AccountOwner::User(owner)),
+            }
+        }
+    }
+}
+
+impl Display for AccountOwner {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let key = match self {
+            AccountOwner::User(owner) => {
+                format!("User:{}", owner)
+            }
+            AccountOwner::Application(app_id) => {
+                format!("Application:{}", app_id)
+            }
+        };
+
+        write!(f, "{}", key)?;
+        Ok(())
+    }
+}
+
+impl FromStr for AccountOwner {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.starts_with("User:") {
+            let owner =
+                Owner::from_str(&s.replace("User:", "")).expect("Getting Owner should not fail");
```

Comment:
You can use `s.strip_prefix("User:")` directly, instead of `starts_with` followed by `replace`.

Reply:
Ah nice! TIL :)